extension/src/plots/model/collect.ts

```diff
@@ -235,33 +238,43 @@ export const collectCheckpointPlotsData = (
     }
   }
 
+  const plotsData: CustomCheckpointPlots = {}
   if (acc.plots.size === 0) {
-    return
+    return plotsData
   }
 
-  const plotsData: CheckpointPlot[] = []
-
   for (const [key, value] of acc.plots.entries()) {
-    plotsData.push({ id: decodeColumn(key), values: value })
+    const decodedMetric = decodeColumn(key)
+    plotsData[decodedMetric] = {
+      id: getCustomPlotId({
+        metric: decodedMetric,
+        type: CustomPlotType.CHECKPOINT
+      }),
+      metric: decodedMetric,
+      type: CustomPlotType.CHECKPOINT,
+      values: value
+    }
   }
 
   return plotsData
 }
 
-export const getCustomPlotId = (metric: string, param: string) =>
-  `custom-${metric}-${param}`
-
-const collectCustomPlotData = (
+const collectMetricVsParamPlotData = (
   metric: string,
   param: string,
   experiments: Experiment[]
-): CustomPlotData => {
+): MetricVsParamPlot => {
   const splitUpMetricPath = splitColumnPath(metric)
   const splitUpParamPath = splitColumnPath(param)
-  const plotData: CustomPlotData = {
-    id: getCustomPlotId(metric, param),
+  const plotData: MetricVsParamPlot = {
+    id: getCustomPlotId({
+      metric,
+      param,
+      type: CustomPlotType.METRIC_VS_PARAM
+    }),
     metric: metric.slice(ColumnType.METRICS.length + 1),
     param: param.slice(ColumnType.PARAMS.length + 1),
```

Comment:
Why/what are we slicing here? I can probably follow the code to know why, but a comment would help. Why aren't we doing it for the id? Can that be a problem later on?

Reply:
> Why/what are we slicing here?

We're taking off the `params:` and `metrics:` from the paths.

>  I can probably follow the code to know why, but a comment would help. Why aren't we doing it for the id? Can that be a problem later on?

The id uses these path prefixes since the `customPlotsOrder` uses them. But you're right, the two different paths are confusing. We can simplify things by having `customPlotsOrder` and `customPlotId` stop using the prefixes.